### PR TITLE
Misc Governance UX Improvements

### PR DESCRIPTION
**Pull request by bcjgit**
# TLDR

- Add Delegate centric vote view
- Add "By line" hover state to indicate Nouns used to propose a prop
- Add hover to Nouns in SQ vote page (i.e. hover over a Noun to see who holds it, when it was born, and link to profile)
- Add hover to Delegate centric vote page (Nouns stacked by delegate address). Indicate address / ENS of delegate and number of Nouns they voted with in given proposal 
- To enable performant Hover states, implement page level (i.e. per`/vote/:id` page on page load) pre-fetching of Noun seeds and ENS resolution which are stored in `localStorage` (ENSs have 30 min TTL to account for ENS changes)

# Testing Done

Tested all states (noun hover card, delegate hover card, by line hover card) on desktop (all that's supported for now) in 
both english and Japanese on a variety of props and results seemed reasonable. 

# Notes to Reviewers
n/a

### CODE REVIEW

**Comment by solimander** [comment]
Starting review in the morning 🌞

File: packages/nouns-webapp/src/pages/Vote/index.tsx [hunk 1]
```diff
@@ -293,24 +296,41 @@ const VotePage = ({
...
+
+        <p
+          onClick={() => setIsDelegateView(!isDelegateView)}
+          className={classes.toggleDelegateVoteView}
+        >
+          {isDelegateView ? (
+            <Trans>Switch to Noun view</Trans>
+          ) : (
+            <Trans>Switch to delegate view</Trans>
+          )}
```
Comment (solimander) [inline packages/nouns-webapp/src/pages/Vote/index.tsx:308]: Oddly, this text is not showing up in the [preview](https://deploy-preview-475--nouns-testnet.netlify.app/vote/83), but does appear locally. Are you seeing the same behavior?

**Comment by solimander** [comment]
May be subgraph-related, but flagging some odd behavior just in case -

Robot Noun being shown, but shows no Noun image and `Voted with 0 Nouns` when hovered (Proposal 90):

sc1

Only one Noun is shown in lower stack despite user having voted with 2 Nouns (Proposal 89):

sc2

**Comment by solimander** [comment]
RE: "Delegated Noun" naming, I suppose they're _technically_ self-delegating, so can ignore that piece if you want.

File: packages/nouns-webapp/src/components/DelegateGroupedNounImageVoteTable/index.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  const shuffledDelegatedGroupedNouns = pseudoRandomPredictableShuffle(
+    filteredDelegateGroupedVoteData,
+    propId,
+  );
+
+  const paddedNounIds = shuffledDelegatedGroupedNouns
+    .map((data: { delegate: string; supportDetailed: 0 | 1 | 2; nounsRepresented: string[] }) => {
+      return (
+        <HoverCard
+          // hoverCardContent={(tip: string) => <>{tip}</>}
```
Comment (solimander) [inline packages/nouns-webapp/src/components/DelegateGroupedNounImageVoteTable/index.tsx:32]: nit: remove

File: packages/nouns-webapp/src/components/NounImageVoteTable/index.tsx [hunk 3]
```diff
@@ -18,7 +20,16 @@ const NounImageVoteTable: React.FC<NounImageVoteTableProps> = props => {
   const shuffledNounIds = pseudoRandomPredictableShuffle(nounIds, propId);
   const paddedNounIds = shuffledNounIds
     .map((nounId: string) => {
-      return <StandaloneNounCircular nounId={EthersBN.from(nounId)} />;
+      return (
+        <HoverCard
+          // hoverCardContent={(tip: string) => <>{tip}</>}
```
Comment (solimander) [inline packages/nouns-webapp/src/components/NounImageVoteTable/index.tsx:25]: nit: remove

File: packages/nouns-webapp/src/components/TightStackedCircleNouns/index.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                    fill: 'none',
+                    stroke: 'white',
+                    strokeWidth: '2',
+                  }}
+                />
+              </clipPath>
+
+              <use xlinkHref={`#${nounIds[i]}`} />
+              <image
+                clip-path={`url(#clipCircleNoun${nounIds[i]})`}
```
Comment (solimander) [inline packages/nouns-webapp/src/components/TightStackedCircleNouns/index.tsx:49]: Use `clipPath` to remedy console error

File: packages/nouns-webapp/src/components/TightStackedCircleNouns/index.tsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+}
+
+// TODO
+const MAX_NOUNS_PER_STACK = 3;
+
+const TightStackedCircleNouns: React.FC<StackedCircleNounsProps> = props => {
+  const { nounIds } = props;
+
+  // TODO not sure if this will be an issue for new nouns or something?
+  const seeds = useNounSeeds();
```
Comment (solimander) [inline packages/nouns-webapp/src/components/TightStackedCircleNouns/index.tsx:16]: This will eventually get populated, but is not guaranteed to exist. Haven't seen that cause any issues yet, however it does seem possible that the `seeds[nounId]` lookup throws because `seeds` is undefined.

File: packages/nouns-webapp/src/components/VoteCard/index.tsx [hunk 6]
```diff
@@ -19,33 +20,73 @@ interface VoteCardProps {
...
+  const [ensCached, setEnsCached] = useState(false);
+
+  // Pre-fetch ENS  of delegates (with 30min TTL)
+  // This makes hover cards load more smoothly
+  useEffect(() => {
+    if (!delegateGroupedVoteData || !library || ensCached) {
+      return;
+    }
+
+    delegateGroupedVoteData.forEach((delegateInfo: { delegate: string }) => {
```
Comment (solimander) [inline packages/nouns-webapp/src/components/VoteCard/index.tsx:68]: Could be improved by only fetching ENS info for delegates that do not exist in local storage.

File: packages/nouns-webapp/src/components/VoteCard/index.tsx [hunk 7]
```diff
@@ -64,7 +105,20 @@ const VoteCard: React.FC<VoteCardProps> = props => {
...
             <Row className={classes.nounProfilePics}>
-              <NounImageVoteTable nounIds={nounIds} propId={parseInt(proposal.id || '0')} />
+              {delegateView ? (
+                <DelegateGroupedNounImageVoteTable
+                  filteredDelegateGroupedVoteData={
+                    delegateGroupedVoteData
+                      ? delegateGroupedVoteData.filter(
+                          v => v.supportDetailed === supportDetailedValue,
+                        )
+                      : []
```
Comment (solimander) [inline packages/nouns-webapp/src/components/VoteCard/index.tsx:115]: nit: cleaner to use optional chaining in these cases -

e.g.

```ts
filteredDelegateGroupedVoteData={
  delegateGroupedVoteData?.filter(
    v => v.supportDetailed === supportDetailedValue,
  ) ?? []
}
```

File: packages/nouns-webapp/src/components/TightStackedCircleNouns/index.tsx (hunk 5 above)
Comment (bcjgit) [reply packages/nouns-webapp/src/components/TightStackedCircleNouns/index.tsx:16]: Added some logic here to hopefully give us just a bit more insurance against this edge case

File: packages/nouns-webapp/src/components/VoteCard/index.tsx [hunk 8]
```diff
@@ -19,54 +22,106 @@ interface VoteCardProps {
...
+        return;
+      }
+
+      library
+        .lookupAddress(delegateInfo.delegate)
+        .then(name => {
+          // Store data as mapping of address_Expiration => address or ENS
+          if (name) {
+            localStorage.setItem(
+              `${delegateInfo.delegate}`,
```
Comment (solimander) [inline packages/nouns-webapp/src/components/VoteCard/index.tsx:80]: Can you add a [cache bucket and use cacheKey](https://github.com/nounsDAO/nouns-monorepo/blob/master/packages/nouns-webapp/src/config.ts#L27-L36) for this, rather than using the address as the top level key in an effort to keep local storage structured?

**Review by solimander — APPROVED** [review]
🚀
